kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -377,10 +390,22 @@ verifyClaimSentence sentence =
             $ sentenceClaimAttributes sentence
         State.modify' $ addClaim (SentenceClaim verified) attrs
   where
-    addClaim verified attrs =
-        Lens.over
-            (field @"indexedModuleClaims")
-            ((attrs, verified) :)
+    addClaim verifiedSentenceClaim@(SentenceClaim verified) attrs =
+        case fromSentenceAxiom (attrs, verified) of
+            Right (OnePathClaimPattern (OnePathRule rulePattern))
+              | not $ containsForall $ left rulePattern
+              , containsForall $ right $ rhs rulePattern ->
+                    error "Found claim with universally-quantified variables \
+                        \appearing only on the right-hand side"
```

Comment:
Since the issue is called "Reject claims..." I only checked the `OnePathClaimPattern` and `AllPathClaimPattern` constructors. Should I also add cases for the others, like `RewriteAxiomPattern`?

Reply:
No, it's alright to have variables that only appear in the right-hand side of a `RewriteAxiomPattern`. We can apply those rules. We want to reject the claims because we can't prove them.

Reply:
That makes sense